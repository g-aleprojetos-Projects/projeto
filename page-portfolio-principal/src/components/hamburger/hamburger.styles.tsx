import styled from 'styled-components';
import colors from 'resourses/colors';
import sizes from 'resourses/sizes';

export interface Props {
  open?: boolean;
  backgroundDark?: boolean;
}

export const Container = styled.button<Props>`
  display: none;
  position: fixed;
  justify-content: space-around;
  flex-flow: column nowrap;
  width: ${sizes.px34};
  height: ${sizes.px34};
  right: 0;
  margin-right: ${sizes.px34};
  margin-top: 40px;
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
  z-index: 20;

  @media (max-width: ${sizes.media_768}) {
    display: flex;
  }
`;

const Span = styled.span<Props>`
  width: ${sizes.px34};
  height: ${sizes.px3};
  background-color: ${props =>
    props.backgroundDark ? colors.white : colors.black};
  border-radius: ${sizes.px8};
  transform-origin: ${sizes.px1};
  transition: all 0.3s linear;
`;

export const SpanSuperior = styled(Span)`
  transform: ${props => (props.open ? 'rotate(45deg)' : 'rotate(0)')};
`;

export const SpanMeio = styled(Span)`
  opacity: ${props => (props.open ? 0 : 1)};
`;

export const SpanInferior = styled(Span)`
  transform: ${props => (props.open ? 'rotate(-45deg)' : 'rotate(0)')};
`;
