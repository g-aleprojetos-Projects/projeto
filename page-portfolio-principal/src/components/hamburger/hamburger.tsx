import React from 'react';
import * as S from './hamburger.styles';
import {useBackgroundContext} from 'context/backgroud';

export interface PropsHumburger {
  openMenu?: boolean;
  mudarEstado: () => void;
}

export const Hamburger = (props: PropsHumburger) => {
  const {temaEscuro: background} = useBackgroundContext();
  const {openMenu = false, mudarEstado} = props;

  return (
    <S.Container data-testid={'test_hamburger'} onClick={mudarEstado}>
      <S.SpanSuperior open={openMenu} backgroundDark={background} />
      <S.SpanMeio open={openMenu} backgroundDark={background} />
      <S.SpanInferior open={openMenu} backgroundDark={background} />
    </S.Container>
  );
};
