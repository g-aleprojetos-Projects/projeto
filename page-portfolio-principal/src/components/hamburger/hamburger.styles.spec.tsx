import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './hamburger.styles';

describe('Hamburger.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();
    expect(container).toMatchSnapshot();
  });

  test('SpanSuperior DEVE ser igual ao snapshot', () => {
    const spanSuperior = renderer.create(<S.SpanSuperior />).toJSON();
    expect(spanSuperior).toMatchSnapshot();
  });

  test('SpanSuperior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for false', () => {
    const spanSuperior = renderer
      .create(<S.SpanSuperior open={true} backgroundDark={false} />)
      .toJSON();
    expect(spanSuperior).toMatchSnapshot();
  });

  test('SpanSuperior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for true', () => {
    const spanSuperior = renderer
      .create(<S.SpanSuperior open={true} backgroundDark={true} />)
      .toJSON();
    expect(spanSuperior).toMatchSnapshot();
  });

  test('SpanMeio DEVE ser igual ao snapshot', () => {
    const spanMeio = renderer.create(<S.SpanMeio />).toJSON();
    expect(spanMeio).toMatchSnapshot();
  });

  test('SpanMeio DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for false', () => {
    const spanMeio = renderer
      .create(<S.SpanMeio open={true} backgroundDark={false} />)
      .toJSON();
    expect(spanMeio).toMatchSnapshot();
  });

  test('SpanMeio DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for true', () => {
    const spanMeio = renderer
      .create(<S.SpanMeio open={true} backgroundDark={true} />)
      .toJSON();
    expect(spanMeio).toMatchSnapshot();
  });

  test('SpanInferior DEVE ser igual ao snapshot', () => {
    const spanInferior = renderer.create(<S.SpanInferior />).toJSON();
    expect(spanInferior).toMatchSnapshot();
  });

  test('SpanInferior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for "false"', () => {
    const spanInferior = renderer
      .create(<S.SpanInferior open={true} backgroundDark={false} />)
      .toJSON();
    expect(spanInferior).toMatchSnapshot();
  });

  test('SpanInferior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for "true"', () => {
    const spanInferior = renderer
      .create(<S.SpanInferior open={true} backgroundDark={true} />)
      .toJSON();
    expect(spanInferior).toMatchSnapshot();
  });
});
