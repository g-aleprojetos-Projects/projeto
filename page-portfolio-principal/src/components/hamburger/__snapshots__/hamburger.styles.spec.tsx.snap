// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: none;
  position: fixed;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  width: 2.125rem;
  height: 2.125rem;
  right: 0;
  margin-right: 2.125rem;
  margin-top: 40px;
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
  z-index: 20;
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<button
  className="c0"
/>
`;

exports[`Hamburger.styles SpanInferior DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<span
  className="c0 c1"
/>
`;

exports[`Hamburger.styles SpanInferior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for "false" 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<span
  className="c0 c1"
  open={true}
/>
`;

exports[`Hamburger.styles SpanInferior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for "true" 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #ffffff;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<span
  className="c0 c1"
  open={true}
/>
`;

exports[`Hamburger.styles SpanMeio DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  opacity: 1;
}

<span
  className="c0 c1"
/>
`;

exports[`Hamburger.styles SpanMeio DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for false 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  opacity: 0;
}

<span
  className="c0 c1"
  open={true}
/>
`;

exports[`Hamburger.styles SpanMeio DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for true 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #ffffff;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  opacity: 0;
}

<span
  className="c0 c1"
  open={true}
/>
`;

exports[`Hamburger.styles SpanSuperior DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<span
  className="c0 c1"
/>
`;

exports[`Hamburger.styles SpanSuperior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for false 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #000000;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<span
  className="c0 c1"
  open={true}
/>
`;

exports[`Hamburger.styles SpanSuperior DEVE ser igual ao snapshot QUANDO "open" for "true" e "backgroumdDark" for true 1`] = `
.c0 {
  width: 2.125rem;
  height: 0.188rem;
  background-color: #ffffff;
  border-radius: 0.5rem;
  -webkit-transform-origin: 0.063rem;
  -ms-transform-origin: 0.063rem;
  transform-origin: 0.063rem;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<span
  className="c0 c1"
  open={true}
/>
`;
